hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -495,14 +496,19 @@ public <T> T doDispatchPrecedingTransaction(
         if (category != TransactionCategory.USER) {
             throw new IllegalArgumentException("Only user-transactions can dispatch preceding transactions");
         }
+
         if (stack.depth() > 1) {
             throw new IllegalStateException(
                     "Cannot dispatch a preceding transaction when a savepoint has been created");
         }
 
-        if (current().isModified()) {
-            throw new IllegalStateException("Cannot dispatch a preceding transaction when the state has been modified");
-        }
+        // This condition fails, because for auto-account creation we charge fees, before dispatching the transaction,
+        // and the state will be modified.
```

Comment:
```suggestion
        // This condition fails, because for auto-account creation we charge fees, before dispatching the transaction, 
        // and the state will be modified. FUTURE: revisit this design
```

Reply:
yes added a TODO